ivy/utils/backend/sub_backend_handler.py

```diff
@@ -200,3 +210,76 @@ def find_available_sub_backends(sub_backends_loc):
             available_sub_backends.append(sub_backend)
 
     return available_sub_backends
+
+
+def _find_available_sub_backend_implementations(sub_backends):
+    result = dict()
+    for sub in sub_backends:
+        sub_backend = ivy.utils.dynamic_import.import_module(_sub_backend_dict[sub])
+        for k, v in sub_backend.__dict__.items():
+            if isinstance(v, Callable) and not k.startswith("__"):
+                result[k] = result.get(k, []) + [sub]
+
+    return result
+
+
+def available_sub_backend_implementations(obj: str) -> list:
+    """
+    Return whether a sub-backend implementation is available for `obj`.
+
+    Parameters
+    ----------
+    obj : str
+        the object for which to check if a sub-backend implementation is available.
+
+    Returns
+    -------
+    ret : list
+        a list of sub-backend implementations available for `obj`.
+
+    Examples
+    --------
+    >>> import ivy
+    >>> ivy.set_backend('torch')
+    >>> ivy.available_sub_backend_implementations("scaled_dot_product_attention")
+    ['xformers']
+    >>> ivy.set_backend('numpy')
+    >>> ivy.available_sub_backend_implementations("scaled_dot_product_attention")
+    []
+    """
+    obj = _check_callable(obj)
```

Comment:
we could just do a `fn = ivy.__dict__[fn_name]` right?
In fact, I think we could simplify this function further so that we don't need `fn` either, we can just import the `sub_backend` dicts and see if `fn_name in sub_backend_dict` for each of the sub-backends rather than iterating over the sub-backend dict which we're doing in `_verify_sub_backends_implementations`.

Reply:
yeah, since we don't check for callable anymore this would be redundant

Reply:
here I'm importing all sub_backends and see if `fn_name in sub_backend_dict` so in the end the `_available_sub_backends_implementations_dict` will be something like
 `{torch: {
        "scaled_dot_product_attention": ["xformers", ...],
        "another_fn": [...],
}`
that's why I'm iterating over all sub backends

Reply:
Sure, but you're also iterating over the `dict` in the `_find_available_sub_backend_implementations` function on this line, I'm referring to loops like `for k, v in sub_backend.__dict__.items():`

Reply:
yeah here I'm pre searching all possible efficient implementations not just for `fn_name` but for all functions available in sub backends and put them in `_available_sub_backends_implementations_dict ` and I only do this  thorough search once for every backend so the next time we want to know if there is an efficient implimentation for a given backend we access  `_available_sub_backends_implementations_dict ` directly without executing `_find_available_sub_backend_implementations`